src/Couchbase.Lite.Shared/Document/DataOps.cs

```diff
@@ -167,6 +169,24 @@ internal static object ToNetObject(object value)
             }
         }
 
+        internal static object ToJsonObject(object value)
+        {
+            switch (value) {
+                case null:
+                    return null;
+                case InMemoryDictionary inMem:
+                    return inMem.ToDictionary();
```

Comment:
How do Blob objects get handled?

Reply:
ToJsonObject(object value) will convert Blob object into json representation dictionary object.

Reply:
I guessed, I couldn't see how this work recursively from the code in the PR. Please make sure if the document / dictionary / array with Blob inside the nested dictionary also works.

For example, the document with the following structure,
{
  "blob: blob1,
   "subDict": { "blob": blob2 }, 
   "subArray": [ blob3 ], 
}

Note: blob1, blob2, and blob3 are the actual Blob objects.

Reply:
I will add more test cases to validate.

Reply:
I added TestBlobJsonStringInLayersOfMutableDict to validate blobs are retriable and values are correct from subarray and sub dictionary in a mutable array.